ecl/hqlcpp/hqlttcpp.cpp

```diff
@@ -10482,6 +10561,37 @@ void HqlScopeTagger::reportError(WarnErrorCategory category, const char * msg)
     errors.report(err);        // will throw immediately if it is an error.
 }
 
+void HqlScopeTagger::reportRootSelectorError(IHqlExpression * expr, IHqlExpression * transformed)
+{
+    HqlExprCopyArray inScope;
+    transformed->gatherTablesUsed(nullptr, &inScope);
+    assertex(inScope.ordinality());
+
+    //Recursively search for an expression which refers to the first selector that is unresolved.
+    IHqlExpression * selector = &inScope.item(0);
+    IHqlExpression * location = nullptr;
+    IHqlExpression * original = findOriginalReference(&location, expr, transformed, selector);
+    if (original)
+    {
+        StringBuffer exprText;
+        getECL(original->queryBody(), exprText);
+
```

Comment:
elideString isn't used here.  Are we sure that this regenerated Ecl not going to be too long?

Reply:
I will change to use elideString, but it would have to be very obscure for this to be too long - the left hand side is almost certainly a named symbol, which will mean the string remains relatively short.  That isn't true for the other output.